### ISSUE TITLE
Categories menu with multiple filter

### ISSUE THREAD

**Issue by in-session** [georgringer/news#625]
Hi Georg,

it is possible to change the categorie menu also multi selecting able?
So after each click on a categorie it will be added of the current categorie: 
Example: &tx_news_pi1[overwriteDemand][categories]=1,5,10

And if the user on &tx_news_pi1[overwriteDemand][categories]=1,5,10 and he clicks on Cat 10 again it will change is to &tx_news_pi1[overwriteDemand][categories]=1,5

Greetings Chris

**Comment by georgringer** [comment]
the only thing which would be missing is adding the category to the current one and a way to unselect categories. all of that should be doable in the template with 1 or 2 additional ViewHelpers. 

I won't add those to the extension but if you need support you can contact me and I will ship those with a custom extension.

**Comment by georgringer** [comment]
Example template for `Category/List.html`

```
<html xmlns:n="http://typo3.org/ns/GeorgRinger/News/ViewHelpers"
	  xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers" data-namespace-typo3-fluid="true">

<f:layout name="General" />
<!--
	=====================
		Templates/Category/List.html
-->

<f:section name="content">
	<f:if condition="{categories}">
		<f:then>


			<f:render section="categoryTree" arguments="{categories:categories,overwriteDemand:overwriteDemand}" />
		</f:then>
		<f:else>
			<f:translate key="list_nocategoriesfound" />
		</f:else>
	</f:if>
</f:section>

<f:section name="categoryTree">
	<ul>
		<f:for each="{categories}" as="category">
			<li>
				<n:multiCategoryLink.isCategoryActive list="{overwriteDemand.categories}" item="{category.item.uid}">
					<f:then>
						<f:link.page title="{category.item.title}" class="active" pageUid="{settings.listPid}"
							additionalParams="{tx_news_pi1:{overwriteDemand:{categories: category.item.uid}}}">{category.item.title}
						</f:link.page>

						(<f:link.page title="{category.item.title}" class="active" pageUid="{settings.listPid}"
									  additionalParams="{n:multiCategoryLink.arguments(mode:'remove',item:category.item.uid,list:overwriteDemand.categories)}">weg
					</f:link.page>)
						
					</f:then>
					<f:else>
						<f:link.page title="{category.item.title}" pageUid="{settings.listPid}"
							additionalParams="{tx_news_pi1:{overwriteDemand:{categories: category.item.uid}}}">{category.item.title}
						</f:link.page>

						(<f:link.page title="{category.item.title}" class="active" pageUid="{settings.listPid}"
									 additionalParams="{n:multiCategoryLink.arguments(mode:'add',item:category.item.uid,list:overwriteDemand.categories)}">dazu
						</f:link.page>)
					</f:else>
				</n:multiCategoryLink.isCategoryActive>

				<f:if condition="{category.children}">
					<f:render section="categoryTree" arguments="{categories: category.children,overwriteDemand:overwriteDemand}" />
				</f:if>
			</li>
		</f:for>
	</ul>
</f:section>
</html>
```

**Comment by enem-frir** [comment]
Hi Georg,

if I change the template there comes an error in the frontend saying: 

> Fluid parse error in template Category_action_list_1594c60a1204dab74fb3cf16ca3b820ec8876928, line 1 at character 1. Error: Unknown Namespace: &lt;n:multiCategoryLink.isCategoryActive list=&quot;{overwriteDemand.categories}&quot; item=&quot;{category.item.uid}&quot;&gt; (error code 0). Template source chunk: 
> 
> TYPO3Fluid\Fluid\Core\Parser\UnknownNamespaceException thrown in file
> /html/typo3/typo3_src-8.7.10/vendor/typo3fluid/fluid/src/Core/Parser/TemplateParser.php in line 157. 

I used your example.
Do you've any idea what this could be?

Greetings Iris

**Comment by marexx** [comment]
Hi, Thanx for the template, works fine.
is it possible to combine Tags and Cats as a Filter? At the moment, Tag List Overrides Cats and also the other way around.
Marek

**Comment by georgringer** [comment]
@marexx that is of course possible. just add `addQueryString="1"` to the links and if you want to remove a setting, add something like `argumentsToBeExcludedFromQueryString="{0:'tx_news_pi1[overwriteDemand][year]'}"`

**Comment by marexx** [comment]
thanx. i solved it, using just categories.
but now i've an other question, 
is it possible to have a filter like:
Find News that have  CAT1 **or** CAT2 **or** CAT3 but not CAT4 **and** not CAT5

**Comment by georgringer** [comment]
yes, by extending it with a custom extension or use something like solr

**Comment by vruescas** [comment]
> thanx. i solved it, using just categories.
> but now i've an other question,
> is it possible to have a filter like:
> Find News that have CAT1 **or** CAT2 **or** CAT3 but not CAT4 **and** not CAT5
> 
> the user should be able to select the categories

You find a solution for this problem? thanks

**Comment by Marat91** [comment]
how can I make checkboxes so that I can select several categories ?
![741d9a1a08](https://user-images.githubusercontent.com/38013831/74833077-d4f7de00-5329-11ea-995c-57bc5fbdacac.png)

**Comment by Marat91** [comment]
![4fbe8d89f8](https://user-images.githubusercontent.com/38013831/74845313-e1873100-533f-11ea-909a-8fccb6f718ac.png)
how do i show selected categories ?

**Comment by Marat91** [comment]
``<f:section name="categoryTree">
	<div class="category__checkbox">
		<f:for each="{categories}" as="category">
		<div class="checkbox-field category__checkbox--field">
			<label>
                <f:if condition="{category.item.uid} == {overwriteDemand.categories}">
                    <f:then>
							<f:form.checkbox
									property="{category.item.title}."
									value="{category.item.title}"
							        class="category__checkbox--input" />
							<span class="checkbox-custom"></span>
							<div class="modal-label">
								<f:link.page title="{category.item.title}" class="active" pageUid="{settings.listPid}"
											 additionalParams="{tx_news_pi1:{overwriteDemand:{categories: category.item.uid}}}">{category.item.title}
								</f:link.page>
							</div>
                    </f:then>
                    <f:else>
							<f:form.checkbox property="{category.item.title}."
								value="{category.item.title}"
								class="category__checkbox--input" />
						<span class="checkbox-custom"></span>
							<div class="modal-label">
								<f:link.page title="{category.item.title}" pageUid="{settings.listPid}"
											 additionalParams="{tx_news_pi1:{overwriteDemand:{categories: category.item.uid}}}">{category.item.title}
								</f:link.page>
							</div>
                    </f:else>
                </f:if>

                <f:if condition="{category.children}">
                    <f:render section="categoryTreeSub" arguments="{categories: category.children,overwriteDemand:overwriteDemand}" />
                </f:if>
			</label>
		</div>
		</f:for>
	</div>

</f:section>``
